tiledb/api/c_api/array_schema/CMakeLists.txt

```diff
@@ -0,0 +1,42 @@
+#
+# tiledb/api/c_api/array_schema/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2024 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+include(common NO_POLICY_SCOPE)
+include(object_library)
+
+list(APPEND SOURCES
+  array_schema_api.cc
+)
+gather_sources(${SOURCES})
+
+commence(object_library capi_array_schema_stub)
+  this_target_sources(${SOURCES})
+  this_target_link_libraries(export)
+  this_target_object_libraries(array_directory array_schema)
```

Comment:
Why is `array_directory` here? That seems off. Specifically, what C API function requires it?

`tiledb_array_schema_load` should really be classified as array function, not as an array schema one. And looking at the C API, it should really take an array handle and not a URI, but that's not something to correct now.

Reply:
> And looking at the C API, [`tiledb_array_schema_load`] should really take an array handle and not a URI

I think the purpose of this API is to read just the schema of an array from storage, without opening the whole array.

Reply:
>the purpose of this API

What this API function does is not the question. The question is whether it belongs in this C API section. I believe it belongs elsewhere.

Separately, using a plain URI rather than an open array is setting the API up for consistency problems in the future. The issue is a race between loading schemas and schema evolution; both should be done on open arrays to ensure consistency.

The point is not to argue the API specifically, but to illustrate that this API function is really an array operation.